code/go/0chain.net/blobbercore/challenge/challenge.go

```diff
@@ -26,186 +24,162 @@ type BCChallengeResponse struct {
 	Challenges []*ChallengeEntity `json:"challenges"`
 }
 
-var cMap = cache.NewLRUCache(2000)
+var lastChallengeTimestamp int
+
+//var cMap = cache.NewLRUCache(2000)
 
 // syncOpenChallenges get challenge from blockchain , and add them in database
 func syncOpenChallenges(ctx context.Context) {
+	const incrOffset = 20
 	defer func() {
 		if r := recover(); r != nil {
 			logging.Logger.Error("[recover]challenge", zap.Any("err", r))
 		}
 	}()
 
+	offset := 0
 	params := make(map[string]string)
 	params["blobber"] = node.Self.ID
-
-	var blobberChallenges BCChallengeResponse
-	blobberChallenges.Challenges = make([]*ChallengeEntity, 0)
-
-	startTime := time.Now()
-	retBytes, err := transaction.MakeSCRestAPICall(transaction.STORAGE_CONTRACT_ADDRESS, "/openchallenges", params, chain.GetServerChain())
-
-	if err != nil {
-		logging.Logger.Error("[challenge]open: ", zap.Error(err))
-		return
-	}
-
-	downloadElapsed := time.Since(startTime)
-
-	bytesReader := bytes.NewBuffer(retBytes)
-	d := json.NewDecoder(bytesReader)
-	d.UseNumber()
-	if err := d.Decode(&blobberChallenges); err != nil {
-		logging.Logger.Error("[challenge]json: ", zap.String("resp", string(retBytes)), zap.Error(err))
-		return
+	params["offset"] = strconv.Itoa(offset)
+	params["limit"] = "20"
+	if lastChallengeTimestamp > 0 {
+		params["from"] = strconv.Itoa(lastChallengeTimestamp)
 	}
+	start := time.Now()
 
-	jsonElapsed := time.Since(startTime)
+	var allOpenChallenges []*ChallengeEntity
 
-	for _, challengeObj := range blobberChallenges.Challenges {
+	var downloadElapsed, jsonElapsed time.Duration
 
-		if challengeObj == nil || challengeObj.ChallengeID == "" {
-			logging.Logger.Info("[challenge]open: No challenge entity from the challenge map")
-			continue
+	for {
```

Comment:
Why use infinite for loop to collect open challenges?
Is it done, so that there won't be too much delay?

What about this situation; You get first request in one second and in other request there was 15 second network delay. This will add delay in the firstly received challenges.
Also we need to make sure that upon each request, we don't successfully get single challenge for say consecutive 10 requests, so that firstly received challenge will fail due to delay.

@peterlimg @sum-it08 @cnlangzi

Reply:
it should be ok. it is designed for pagination.

Reply:
we should timeout API calls in `makeSCRestAPICall` then
to be addressed as a separate issue.